frontend/src/modules/EditZing/Components/ExportFileIconButton.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { ReactComponent as CSVExportImg } from '@assets/img/csvexport.svg'
+import { ExportFileIconButtonType } from 'EditZing/Types/ComponentProps'
+import { Button } from '@material-ui/core'
+import { CSV_FILE, DOWNLOAD_ALL, DOWNLOAD_NETIDS } from '@core'
+import { CSVLink } from 'react-csv'
+import { Data } from 'react-csv/components/CommonPropTypes'
+
+// converts contents of the button depending on the type
+export const ExportFileIconButton = ({
+  type,
+  data,
+  downloadData,
+}: ExportFileIconButtonType) => {
+  // fix the typing for data here?
```

Comment:
Idk what I am reading here lol

Reply:
What are you referring to?

Reply:
If you meant fix the typing as in `generateData: { (): any[]; (): string | Data }` I find that a lil confusing but this is not a very important comment

Reply:
Oh yeah lol I had WebStorm generate that for me bc I was too lazy to write the type out but I will try and fix it so that it is not as confusing.